### PR TITLE
Codeception/Symfony: Fix deprecations

### PR DESCRIPTION
**Pull request by blankse**
Deprecation from Codeception log:
```
DEPRECATION: Method "Pimcore\Model\Listing\AbstractListing::isValidOrderKey()" might add "bool" as a native return type declaration in the future. Do the same in child class "CustomerManagementFrameworkBundle\Model\ActionTrigger\Rule\Listing" now to avoid errors or add an explicit @return annotation to suppress this message. /home/runner/work/customer-data-framework/customer-data-framework/vendor/symfony/error-handler/DebugClassLoader.php:326
DEPRECATION: Method "Pimcore\Model\Listing\Dao\AbstractDao::load()" might add "array" as a native return type declaration in the future. Do the same in child class "CustomerManagementFrameworkBundle\Model\ActionTrigger\Rule\Listing\Dao" now to avoid errors or add an explicit @return annotation to suppress this message. /home/runner/work/customer-data-framework/customer-data-framework/vendor/symfony/error-handler/DebugClassLoader.php:326
DEPRECATION: Method "Pimcore\Model\Listing\AbstractListing::isValidOrderKey()" might add "bool" as a native return type declaration in the future. Do the same in child class "CustomerManagementFrameworkBundle\Model\ActivityList\DefaultMariaDbActivityList" now to avoid errors or add an explicit @return annotation to suppress this message. /home/runner/work/customer-data-framework/customer-data-framework/vendor/symfony/error-handler/DebugClassLoader.php:326
```

Other deprecations are from the Symfony Toolbar

### CODE REVIEW

File: src/Security/UserProvider/CustomerObjectUserProvider.php [hunk 1]
```diff
@@ -74,26 +76,32 @@ public function loadUserByUsername($username)
 
     /**
      * @inheritdoc
+     *
+     * @return UserInterface
      */
-    public function refreshUser(UserInterface $user)
+    public function refreshUser(UserInterface $user)//: UserInterface
```
Comment (dvesh3) [inline src/Security/UserProvider/CustomerObjectUserProvider.php]:
```suggestion
    public function refreshUser(UserInterface $user)
```
Return type hint not added on Symfony 6

File: src/Security/UserProvider/CustomerObjectUserProvider.php [hunk 2]
```diff
@@ -74,26 +76,32 @@ public function loadUserByUsername($username)
...
+        return $customer;
     }
 
     /**
      * @inheritdoc
+     *
+     * @return bool
      */
-    public function supportsClass($class)
+    public function supportsClass($class)//: bool
```
Comment (dvesh3) [inline src/Security/UserProvider/CustomerObjectUserProvider.php]:
```suggestion
    public function supportsClass($class)
```
Return type hint not added on Symfony 6

File: src/Security/UserProvider/CustomerObjectUserProvider.php (hunk 1 above)
Comment (blankse) [reply src/Security/UserProvider/CustomerObjectUserProvider.php]: But shouldn't we add it anyway? It could come in version 7.

Comment (dvesh3) [reply src/Security/UserProvider/CustomerObjectUserProvider.php]: I avoid adding this now so that we don't accidently enable it for Symfony 6.

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@blankse thanks!

**Pull request merged**